### PR TITLE
Implement Userlist View 

### PR DESCRIPTION
**Pull request by ifndefdeadmau5**
This pull request is about only implementing view of user list.
Feature fetching data from server doesn't included.

We can ignore all of generated code except below files.

+8 −2  src/components/App.js
+8 −6  src/components/Login.js
+1 −1  src/components/SplashPage.js
+25 −0  src/components/UserList/Footer.js
+36 −0  src/components/UserList/Header.js
+64 −0  src/components/UserList/Row.js
+79 −0  src/components/UserList/UserList.js

### CODE REVIEW

File: src/components/Login.js [hunk 1]
```diff
@@ -95,6 +96,10 @@ class Login extends Component {
                         AccessToken.getCurrentAccessToken().then(
                           (data) => {
                             alert(data.accessToken.toString());
```
Comment (ethkim) [inline src/components/Login.js:101]: This alert thing is only for debug purposes, right? Please add a TODO to remove this before we go production.

File: src/components/UserList/UserList.js [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+ View,
+ Navigator,
+ Image,
+ ListView,
+} from 'react-native';
+import Header from './Header';
+import Footer from './Footer';
+import Row from './Row';
+
+const API_KEY = '73b19491b83909c7e07016f4bb4644f9:2:60667290';
```
Comment (ethkim) [inline src/components/UserList/UserList.js]: Hm. Could you point out where this API_KEY goes into each native code? We should probably encrypt the key for security purposes.

File: src/components/UserList/UserList.js [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+const QUERY_TYPE = 'hardcover-fiction';
```
Comment (ethkim) [inline src/components/UserList/UserList.js]: What's a hardcover-fiction?

File: src/components/UserList/UserList.js [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+const API_STEM = 'http://api.nytimes.com/svc/books/v3/lists';
```
Comment (ethkim) [inline src/components/UserList/UserList.js]: ??

File: src/components/UserList/UserList.js [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+    return <Row />;
+  }
+
+  // Refresh listview
+  _refreshData() {
+    fetch(ENDPOINT)
+      .then((response) => response.json())
+      .then((rjson) => {
+        this.setState({
+          dataSource: this.state.dataSource.cloneWithRows(rjson.results.books),
```
Comment (ethkim) [inline src/components/UserList/UserList.js]: I think we need to break the line here somewhere.

File: src/components/UserList/UserList.js [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+import Header from './Header';
+import Footer from './Footer';
+import Row from './Row';
+
+const API_KEY = '73b19491b83909c7e07016f4bb4644f9:2:60667290';
+const QUERY_TYPE = 'hardcover-fiction';
+const API_STEM = 'http://api.nytimes.com/svc/books/v3/lists';
+const ENDPOINT = `${API_STEM}/${QUERY_TYPE}?response-format=json&api-key=${API_KEY}`;
+
+class UserList extends Component {
```
Comment (ethkim) [inline src/components/UserList/UserList.js:14]: Could we comment on each class / methods in this PR?

File: src/components/UserList/UserList.js [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+  constructor(props) {
+    super(props);
+    let dataSource = new ListView.DataSource({
+      rowHasChanged: (row1, row2) => row1 != row2,
+    });
+    this.state = {
+      dataSource: dataSource.cloneWithRows([1, 2, 3, 4, 5]),
```
Comment (ethkim) [inline src/components/UserList/UserList.js]: What is this [1,2,3,4,5]? Is this duplicating the mock Row object above? More comments please!

File: src/components/UserList/Row.js [hunk 8]
```diff
@@ -0,0 +1,67 @@
...
+
+class Row extends Component {
+  render() {
+    return (
+      <View style={styles.row}>
+        <Image style={styles.photo}
+          source={{ uri: 'http://lorempixel.com/64/64/cats/' }} />
+        <View style={styles.imageSeperator}></View>
+        <View style={styles.userInformation}>
+          <Text style={styles.name}>Hyunchan Kim</Text>
```
Comment (ethkim) [inline src/components/UserList/Row.js]: This seems like a mock data. Could you add a comment/TODO here to replace this with real data from backend?

**Comment by ethkim** [comment]
Thanks Seongmin!

I'm not sure what's the best way to test this. I know you guys are investigating into Jest - until that's in place, are you guys actually testing it with real devices, or is it just being tested on emulators?

File: src/components/Login.js (hunk 1 above)
Comment (ace0625) [reply src/components/Login.js:101]: Yes. It's only for debugging. Will add a Todo and be updated by tomorrow.

File: src/components/UserList/UserList.js (hunk 4 above)
Comment (ifndefdeadmau5) [reply src/components/UserList/UserList.js]: These constants were for networking test purpose. It'll be replaced our back-end API endpoint. will add comment!

File: src/components/UserList/UserList.js (hunk 6 above)
Comment (ifndefdeadmau5) [reply src/components/UserList/UserList.js:14]: Okay!

File: src/components/UserList/UserList.js (hunk 5 above)
Comment (ifndefdeadmau5) [reply src/components/UserList/UserList.js]: Hmm... hard to find appropriate point.... but okay!

File: src/components/UserList/UserList.js (hunk 7 above)
Comment (ifndefdeadmau5) [reply src/components/UserList/UserList.js]: you're right~

**Comment by ifndefdeadmau5** [comment]
Currently we are testing on real devices.

File: src/components/UserList/UserList.js [hunk 9]
```diff
@@ -1,29 +1,28 @@
...
+    // Method 'rowHasChanged' must be implemented to use listview.
     let dataSource = new ListView.DataSource({
-      rowHasChanged: (row1, row2) => row1 != row2,
+      rowHasChanged: (row1, row2) => row1 !== row2,
     });
     this.state = {
-      dataSource: dataSource.cloneWithRows([1, 2, 3, 4, 5]),
+
+      // Initialize with mock data for testing listview.
+      dataSource: dataSource.cloneWithRows([1, 2, 3, 4, 5,1, 2, 3, 4, 5,1, 2, 3, 4, 5,1, 2, 3, 4, 5,1, 2, 3, 4, 5,1, 2, 3, 4, 5,1, 2, 3, 4, 5]),
```
Comment (ethkim) [inline src/components/UserList/UserList.js:25]: break line at 100 chars per line. 
There are no spaces between 5, and 1.

File: src/components/UserList/UserList.js [hunk 10]
```diff
@@ -35,44 +34,50 @@ class UserList extends Component {
...
+    //     this.setState({
+    //       dataSource:
+    // this.state.dataSource.cloneWithRows(rjson.results),
+    //     });
+    //   });
   }
 
   render() {
     return (
+<<<<<<< Updated upstream
```
Comment (ethkim) [inline src/components/UserList/UserList.js]: Merge issue? Please fix.

**Comment by ethkim** [comment]
Took another pass. Please see comments. Thanks!

**Comment by ifndefdeadmau5** [comment]
Oops.. I didn't check after applying stash(merge) at previous commit. I'll fix this!

**Comment by ifndefdeadmau5** [comment]
LGTM

**Pull request merged**